fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/impl/BlankNodeRdfContext.java

```diff
@@ -1,122 +0,0 @@
-/**
- * Copyright 2015 DuraSpace, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.fcrepo.kernel.modeshape.rdf.impl;
-
-import com.google.common.base.Function;
-import com.google.common.base.Predicate;
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Iterators;
-import com.hp.hpl.jena.rdf.model.Resource;
-
-import org.fcrepo.kernel.api.models.FedoraResource;
-import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
-import org.fcrepo.kernel.api.identifiers.IdentifierConverter;
-import org.fcrepo.kernel.api.utils.iterators.RdfStream;
-import org.fcrepo.kernel.modeshape.rdf.impl.mappings.PropertyValueIterator;
-
-import javax.jcr.Node;
-import javax.jcr.Property;
-import javax.jcr.RepositoryException;
-import javax.jcr.Value;
-
-import java.util.Iterator;
-import java.util.List;
-
-import static java.util.Arrays.asList;
-import static javax.jcr.PropertyType.PATH;
-import static javax.jcr.PropertyType.REFERENCE;
-import static javax.jcr.PropertyType.WEAKREFERENCE;
-import static org.fcrepo.kernel.modeshape.identifiers.NodeResourceConverter.nodeConverter;
-import static org.fcrepo.kernel.modeshape.utils.FedoraTypesUtils.isBlankNode;
-
-/**
- * Embed all blank nodes in the RDF stream
- *
- * @author cabeer
- * @author ajs6f
- * @since 10/9/14
- */
-public class BlankNodeRdfContext extends NodeRdfContext {
```

Comment:
I do not believe that `git mv` will preserve the history of this class. Git uses its own heuristic for determining whether a file has been deleted and added somewhere else (e.g. a simple "move") or whether a newly added file is a truly different thing (e.g. a delete and an unrelated addition). If you look at the implementation of `SkolemNodeRdfContext`, it shares very few similar lines with `BlankNodeRdfContext`. So, even with `git mv`, that history is not necessarily going to be saved. The only sure way to keep that history is to rename the file but keep the implementation basically intact (as much as possible) in one commit. In a subsequent commit (that does not get squashed) the implementation can be changed to reflect the Java8 idioms as expressed here.

Personally, I'm not convinced that doing this simply to preserve the history (first a rename with minimal changes to the file followed by a subsequent, more substantive reimplementation) is worth the effort.